aten/src/ATen/native/native_functions.yaml

```diff
@@ -3358,93 +3980,367 @@
   variants: method, function
 
 - func: renorm(Tensor self, Scalar p, int dim, Scalar maxnorm, *, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
+    CPU: renorm_out_cpu
+    CUDA: renorm_out_cuda
 
 - func: renorm(Tensor self, Scalar p, int dim, Scalar maxnorm) -> Tensor
   variants: method, function
+  dispatch:
+    CPU: renorm_cpu
+    CUDA: renorm_cuda
 
 - func: unfold(Tensor(a) self, int dimension, int size, int step) -> Tensor(a)
+  cpu_half: True
+  cpu_bool: True
+  cuda_bool: True
   variants: method
+  dispatch:
+    CPU: unfold_cpu
+    CUDA: unfold_cuda
 
 - func: equal(Tensor self, Tensor other) -> bool
+  cpu_bool: True
   variants: method, function
+  dispatch:
+    CPU: equal_cpu
+    CUDA: equal_cuda
 
 - func: pow(Tensor self, Tensor exponent, *, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
+    CPU: pow_out_cpu
+    CUDA: pow_out_cuda
 
 - func: pow(Tensor self, Tensor exponent) -> Tensor
   variants: method, function
+  dispatch:
+    CPU: pow_cpu
+    CUDA: pow_cuda
 
 - func: pow(Scalar self, Tensor exponent, *, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
+    CPU: pow_out_cpu
+    CUDA: pow_out_cuda
 
 - func: pow(Scalar self, Tensor exponent) -> Tensor
+  dispatch:
+    CPU: pow_cpu
+    CUDA: pow_cuda
 
 - func: normal(Tensor mean, float std=1, *, Generator? generator=None, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
+    CPU: normal_out_cpu
+    CUDA: normal_out_cuda
 
 - func: normal(Tensor mean, float std=1, *, Generator? generator=None) -> Tensor
+  dispatch:
+    CPU: normal_cpu
+    CUDA: normal_cuda
 
 - func: normal(float mean, Tensor std, *, Generator? generator=None, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
+    CPU: normal_out_cpu
+    CUDA: normal_out_cuda
 
 - func: normal(float mean, Tensor std, *, Generator? generator=None) -> Tensor
+  dispatch:
+    CPU: normal_cpu
+    CUDA: normal_cuda
 
 - func: normal(Tensor mean, Tensor std, *, Generator? generator=None, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
+    CPU: normal_out_cpu
+    CUDA: normal_out_cuda
 
 - func: normal(Tensor mean, Tensor std, *, Generator? generator=None) -> Tensor
+  dispatch:
+    CPU: normal_cpu
+    CUDA: normal_cuda
 
 - func: alias(Tensor(a) self) -> Tensor(a)
+  cpu_half: True
+  cpu_bool: True
+  cuda_bool: True
   variants: method, function
+  dispatch:
+    CPU: alias_cpu
+    CUDA: alias_cuda
 
 - func: _dirichlet_grad(Tensor x, Tensor alpha, Tensor total, *, Tensor(a!) out) -> Tensor(a!)
+  dispatch:
+    CPU: _dirichlet_grad_out_cpu
 
 - func: _dirichlet_grad(Tensor x, Tensor alpha, Tensor total) -> Tensor
+  dispatch:
+    CPU: _dirichlet_grad_cpu
+
+- func: _getri_single(Tensor self) -> Tensor
+  dispatch:
+    CPU: _getri_single_cpu
+    CUDA: _getri_single_cuda
+
+- func: _index_copy_(Tensor(a!) self, int dim, Tensor index, Tensor source) -> Tensor(a!)
```

Comment:
can we make some more effort to reduce the number of these new functions:
1) all of the linear algebra methods are trivially done, just move the 1d checks into the functions themselves.  It wasn't done that way originally because we couldn't tell the difference between 1-d and 0-d tensors in TH.  This should be a separate PR.
2) min/max/copy_ignoring_overlaps looks like it's only ever used in the context where you know the dispatch device.
3) mode doesn't seem to be used

Reply:
1. I've moved all of them except addr, which I can't do without making error messages worse.
2. min and max must be wrapped because while we only call them when we know it's cuda, it's in a function that serves cpu and cuda. The cuda version doesn't exist when we compile for cpu only, so the dispatch is needed.
3. mode is used in the same file as min and max